Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm

```diff
@@ -1567,9 +1569,11 @@ static URL conformFragmentIdentifierForURL(const URL& url)
     CMTime cmBefore = PAL::toCMTime(negativeTolerance);
     CMTime cmAfter = PAL::toCMTime(positiveTolerance);
 
-    // [AVPlayerItem seekToTime] will throw an exception if toleranceBefore is negative.
-    if (PAL::CMTimeCompare(cmBefore, PAL::kCMTimeZero) < 0)
-        cmBefore = PAL::kCMTimeZero;
+    // [AVPlayerItem seekToTime] will throw an exception if a tolerance is invalid or negative.
+    if (!CMTIME_IS_VALID(cmBefore) || PAL::CMTimeCompare(cmBefore, PAL::kCMTimeZero) < 0)
+        cmBefore = PAL::kCMTimePositiveInfinity;
```

Comment:
I wonder if we should be clamping to zero when given a negative or invalid time, rather than infinity. I don't object to the proposed behavior per se. Hopefully the added assertions in HTMLMediaElement will help catch the case where the tolerances are mal-formed.